src/test/java/org/cryptomator/cryptofs/health/type/CiphertextFileTypeCheckTest.java

```diff
@@ -0,0 +1,227 @@
+		}
+
+		@Test
+		@DisplayName("tests if dirs ending with c9s are visited and closer looked at")
+		public void testC9sDirsAreVisited() throws IOException {
+			Path p = dataRoot.resolve("AA/zzzz/bar=.c9s");
+			Files.createDirectories(p);
+
+			var visitorSpy = Mockito.spy(visitor);
+			Files.walkFileTree(dataRoot, Set.of(), 3, visitorSpy);
+
+			Mockito.verify(visitorSpy).visitFile(Mockito.eq(p), Mockito.any());
+			Mockito.verify(visitorSpy).checkCiphertextTypeC9s(p);
+			Mockito.verify(visitorSpy, Mockito.never()).checkCiphertextTypeC9r(p);
+		}
+
+		@Nested
+		@DisplayName("Ciphertext type checks for c9r dir")
+		public class C9rDirChecked {
+
+			private Path c9rDir;
+
+			@BeforeEach
+			public void initialize() throws IOException {
+				c9rDir = dataRoot.resolve("AA/pretended/c9rDir");
+				Files.createDirectories(c9rDir);
+			}
+
+			@ParameterizedTest
+			@DisplayName("c9r dir with only dir.c9r or symlink.c9r is known")
+			@ValueSource(strings = {"dir.c9r", "symlink.c9r"})
+			public void testSigDirOrSymlinkFileProduceKnownType(String signatureFile) throws IOException {
+				Files.createFile(c9rDir.resolve(signatureFile));
+
+				visitor.checkCiphertextTypeC9r(c9rDir);
+
+				ArgumentCaptor<DiagnosticResult> resultCaptor = ArgumentCaptor.forClass(DiagnosticResult.class);
+				Mockito.verify(resultsCollector).accept(resultCaptor.capture());
+				MatcherAssert.assertThat(resultCaptor.getValue(), Matchers.instanceOf(KnownType.class));
+			}
+
+			@Test
+			@DisplayName("c9r dir with dir.c9r AND symlink.c9r is ambiguous")
+			public void testDirC9rAndSymlinkC9rProducesAmbiguousType() throws IOException {
+				Files.createFile(c9rDir.resolve("dir.c9r"));
+				Files.createFile(c9rDir.resolve("symlink.c9r"));
+
+				visitor.checkCiphertextTypeC9r(c9rDir);
+
+				ArgumentCaptor<DiagnosticResult> resultCaptor = ArgumentCaptor.forClass(DiagnosticResult.class);
+				Mockito.verify(resultsCollector).accept(resultCaptor.capture());
+				MatcherAssert.assertThat(resultCaptor.getValue(), Matchers.instanceOf(AmbiguousType.class));
+			}
+
+			@Test
+			@DisplayName("c9r dir without dir.c9r or symlink.c9r is unknown")
+			public void testNoneSignatureFileProducesUnknownType() throws IOException {
+				Path p = dataRoot.resolve("AA/aaaa/zaz.c9r");
+				Files.createDirectories(p);
+
+				visitor.checkCiphertextTypeC9r(p);
+
+				ArgumentCaptor<DiagnosticResult> resultCaptor = ArgumentCaptor.forClass(DiagnosticResult.class);
+				Mockito.verify(resultsCollector).accept(resultCaptor.capture());
+				MatcherAssert.assertThat(resultCaptor.getValue(), Matchers.instanceOf(UnknownType.class));
+			}
+
+			@ParameterizedTest
+			@DisplayName("contents.c9r is ignored for c9r dir")
+			@MethodSource("provideAllContentsC9rCombos")
+			public void testContentsC9rIsIgnored(List<String> sigFiles, Class resultClass) throws IOException {
+				for (var sigFile : sigFiles) {
+					Files.createFile(c9rDir.resolve(sigFile));
+				}
```

Comment:
`sigFiles`?

Reply:
sigFiles = signatureFiles

renamed in f3c94769af0aedb0c738d37f33cb0c2e2a608d13

Reply:
jup I saw signatureFiles elsewhere, however I am wondering what this has to do with signatures?

Reply:
When I used the term i had "signature file" as in "signature move" in mind: An identifying file.

Reply:
Oh ok, got confused due to the cryptography context. Let's stick to "type" used elsewhere, i.e. `typeFiles`

Reply:
Fixed in 765c011